### DISCUSSION TITLE
Feature Switches in Typer

### DISCUSSION THREAD

**Discussion by archiif** [fastapi/typer#1322] [Questions]
### First Check

- [X] I added a very descriptive title to this issue.
- [X] I used the GitHub search to find a similar issue and didn't find it.
- [X] I searched the Typer documentation, with the integrated search.
- [X] I already searched in Google "How to X in Typer" and didn't find any information.
- [X] I already read and followed all the tutorial in the docs and didn't find an answer.
- [X] I already checked if it is not related to Typer but to [Click](https://github.com/pallets/click).

### Commit to Help

- [X] I commit to help with one of those options 👆

### Example Code

```python
import sys
import click

@click.command()
@click.option('--upper', 'transformation', flag_value='upper',
              default=True)
@click.option('--lower', 'transformation', flag_value='lower')
def info(transformation):
    click.echo(getattr(sys.platform, transformation)())
```

### Description

Is it possible to have multiple options map to a single parameter?
It's one of the basic patterns possible in Click, called Feature Switches: https://click.palletsprojects.com/en/8.0.x/options/#feature-switches
Given Typer's 1-to-1 option-to-parameter mapping, would this feature be impossible to implement in Typer? If not, what would the equivalent Typer code for the Click code above look like?

### Operating System

Windows

### Typer Version

0.4.0

### Python Version

Python 3.10.0

**Comment by YuriiMotov** [comment]
There is no such feature in Typer, but if you really need such behavior I think you can achieve this using callback and global state object:

```py
from dataclasses import dataclass
from typing import Optional

import typer
from typing_extensions import Annotated

@dataclass
class State:
    feature_switch: str = "undefined"

state = State()

def feature_switch_callback(ctx: typer.Context, param: typer.CallbackParam, value: bool):
    if value:
        # You can add logic to make these flags mutually exclusive
        state.feature_switch = param.name
    return value

def main(
    upper: Annotated[Optional[bool], typer.Option("--upper", callback=feature_switch_callback)] = None,
    lower: Annotated[Optional[bool], typer.Option("--lower", callback=feature_switch_callback)] = None
):
    typer.echo(state.feature_switch)

if __name__ == "__main__":
    typer.run(main)
```

Try it:

```
$ python main.py --upper
upper

$ python main.py --lower
lower

$ python main.py --lower --upper
upper

$ python main.py
undefined
```
